src/dioptra/restapi/v1/artifacts/schema.py

```diff
@@ -69,6 +96,26 @@ class ArtifactSchema(ArtifactMutableFieldsSchema, ArtifactBaseSchema):  # type:
         metadata=dict(description="URL pointing to the location of the Artifact."),
         required=True,
     )
+    isDir = fields.Boolean(
+        attribute="is_dir",
+        metadata=dict(
+            description="Indicates whether or not the Artifact is a directory."
+        ),
+        required=True,
+    )
+    files = fields.Nested(
+        ArtifactFileSchema,
+        attribute="files",
+        metadata=dict(description="The list of files contained in an Artifact."),
+        required=True,
+    )
+    fileTree = fields.Nested(
+        ArtifactFileSchema,
```

Comment:
need a schema for the file tree view. I'm not sure if we defined this yet. I'm also not sure how it would work with the paging parameters.

The intent of the file tree was to provide the frontend with a nested file tree structure that makes it easy to display in a file browser like UI. If we are concerned with the amount of data that might be returned and need paging, maybe we can instead remove `fileTree` and have the UI use `files`, but add a `recurse` boolean query param that determines if we want a flat list of all files, or if we just want the files in the directory in the `path` query param. This way the UI would work by making a call to `/artifacts/{id}/contents?path=PATH&recurse=False`, updating the `path` each time the user clicks a file/directory in the browser.